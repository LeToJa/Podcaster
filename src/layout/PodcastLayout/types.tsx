import { RSSResponse } from '../../helpers/RSSParser/types'
import { ReactElementWithChildrenTypes } from '../../types'

export interface PodcastLayoutTypes extends ReactElementWithChildrenTypes {
    aside: PodcastAsideTypes
}

export interface PodcastAsideTypes {
    id: string
    title: string
    author: string
    artwork: string
    description: string
}

export interface PodcastItemsTypes extends PodcastAsideTypes {
    episodes: RSSResponse[] | false
}