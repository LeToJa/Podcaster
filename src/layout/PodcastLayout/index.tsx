import { Link } from 'react-router-dom'
import { Parser } from 'html-to-react'

import { PodcastLayoutTypes } from './types'

const PodcastLayout = ({ children, aside: { id, title, author, artwork, description } }: PodcastLayoutTypes) => <>
    <section className='grid grid-cols-7 gap-12'>
        <aside className='col-span-2'>
            <figure className='block shadow-xl p-4'>
                <img className='w-4/5 mx-auto rounded-full' src={artwork} alt={title} />
                <header className='p-4'>
                    <Link to={`/podcast/${id}`}>
                        <h4 className='font-bold text-xl mb-2'>{title}</h4>
                        <h6 className='text-gray-700 text-base'>{author}</h6>
                    </Link>
                </header>
                <blockquote className='text-sm'>
                    <h5 className='font-bold text-lg mb-2'>Description</h5>
                    {Parser().parse(description)}
                </blockquote>
            </figure>
        </aside>
        <article className='col-span-5'>
            {children}
        </article>
    </section>
</>

export default PodcastLayout