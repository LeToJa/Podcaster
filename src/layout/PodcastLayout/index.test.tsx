import { render } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import { BrowserRouter } from 'react-router-dom'
import PodcastLayout from '.'


const mockProps = {
    id: '123',
    title: 'Sample Podcast',
    author: 'John Doe',
    artwork: 'https://via.placeholder.com/150',
    description: '<p>This is a <strong>sample</strong> podcast description.</p>',
}

describe('PodcastLayout', () => {
    it('renders correctly and matches snapshot', () => {
        const { container } = render(
            <BrowserRouter>
                <PodcastLayout aside={mockProps}>
                    <figure>Podcast Episode List</figure>
                </PodcastLayout>
            </BrowserRouter>
        )

        expect(container).toMatchSnapshot()
    })

    it('renders with correct content and link', () => {
        const { getByAltText, getByText } = render(
            <BrowserRouter>
                <PodcastLayout aside={mockProps}>
                    <figure>Podcast Episode List</figure>
                </PodcastLayout>
            </BrowserRouter>
        )

        expect(getByAltText('Sample Podcast')).toBeTruthy()
        expect(getByText('Sample Podcast')).toBeTruthy()
        expect(getByText('John Doe')).toBeTruthy()
        expect(getByText('Sample Podcast').closest('a')).toHaveProperty('href', 'http://localhost:3000/podcast/123')
        expect(getByText('podcast description', { exact: false })).toBeTruthy()
        expect(getByText('Podcast Episode List')).toBeTruthy()
    })
})
