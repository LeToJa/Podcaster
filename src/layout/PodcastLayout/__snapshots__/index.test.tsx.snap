// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PodcastLayout > renders correctly and matches snapshot 1`] = `
<div>
  <section
    class="grid grid-cols-7 gap-12"
  >
    <aside
      class="col-span-2"
    >
      <figure
        class="block shadow-xl p-4"
      >
        <img
          alt="Sample Podcast"
          class="w-4/5 mx-auto rounded-full"
          src="https://via.placeholder.com/150"
        />
        <header
          class="p-4"
        >
          <a
            href="/podcast/123"
          >
            <h4
              class="font-bold text-xl mb-2"
            >
              Sample Podcast
            </h4>
            <h6
              class="text-gray-700 text-base"
            >
              John Doe
            </h6>
          </a>
        </header>
        <blockquote
          class="text-sm"
        >
          <h5
            class="font-bold text-lg mb-2"
          >
            Description
          </h5>
          <p>
            This is a 
            <strong>
              sample
            </strong>
             podcast description.
          </p>
        </blockquote>
      </figure>
    </aside>
    <article
      class="col-span-5"
    >
      <figure>
        Podcast Episode List
      </figure>
    </article>
  </section>
</div>
`;
